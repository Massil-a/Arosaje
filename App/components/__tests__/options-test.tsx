import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import OptionsScreen from '../../app/(tabs)/options';
import AsyncStorage from '@react-native-async-storage/async-storage';
jest.mock('@react-native-async-storage/async-storage', () => ({
    setItem: jest.fn(),
    getItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
}));

describe('OptionsScreen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <OptionsScreen />
      </NavigationContainer>
    );

    // Générer le snapshot
    expect(toJSON()).toMatchSnapshot();
  });

  it('opens modal on Deconnecter press', () => {
    const { getByText } = render(
      <NavigationContainer>
        <OptionsScreen />
      </NavigationContainer>
    );

    // Simuler un clic sur le bouton "Déconnecter"
    const deconnecterButton = getByText('Déconnecter');
    fireEvent.press(deconnecterButton);

  });

  it('closes modal on modal close action', () => {
    const { getByText } = render(
      <NavigationContainer>
        <OptionsScreen />
      </NavigationContainer>
    );

    // Simuler un clic sur le bouton "Déconnecter"
    const deconnecterButton = getByText('Déconnecter');
    fireEvent.press(deconnecterButton);

    // Vérifier que la modal est ouverte
    expect(getByText('Déconnecter')).toBeTruthy(); // Assurez-vous que le texte correspond à votre modal

  });

  it('navigates to the correct screen on button press', () => {
    const { getByText } = render(
      <NavigationContainer>
        <OptionsScreen />
      </NavigationContainer>
    );

    // Simuler un clic sur le bouton "Informations personnelles"
    const infoButton = getByText('Informations personnelles');
    fireEvent.press(infoButton);

  });
});
