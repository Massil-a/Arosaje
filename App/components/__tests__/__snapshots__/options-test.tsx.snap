// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "marginTop": 30,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "color": "#BDBDBD",
                      "padding": 10,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    Informations personnelles
                  </Text>
                  <Text />
                </View>
              </View>
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#E8E8E8",
                    "height": 1,
                    "marginVertical": 5,
                    "width": "95%",
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "color": "#BDBDBD",
                      "padding": 10,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    Sécurité du compte
                  </Text>
                  <Text />
                </View>
              </View>
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#E8E8E8",
                    "height": 1,
                    "marginVertical": 5,
                    "width": "95%",
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "color": "#BDBDBD",
                      "padding": 10,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    Notifications
                  </Text>
                  <Text />
                </View>
              </View>
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#E8E8E8",
                    "height": 1,
                    "marginVertical": 5,
                    "width": "95%",
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "color": "#BDBDBD",
                      "padding": 10,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    Questions
                  </Text>
                  <Text />
                </View>
              </View>
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#E8E8E8",
                    "height": 1,
                    "marginVertical": 5,
                    "width": "95%",
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "color": "#BDBDBD",
                      "padding": 10,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    Mes données
                  </Text>
                  <Text />
                </View>
              </View>
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#E8E8E8",
                    "height": 1,
                    "marginVertical": 5,
                    "width": "95%",
                  }
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "color": "#BDBDBD",
                      "padding": 10,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "fontSize": 14,
                      }
                    }
                  >
                    Informations légales
                  </Text>
                  <Text />
                </View>
              </View>
              <View
                style={
                  {
                    "alignSelf": "center",
                    "backgroundColor": "#E8E8E8",
                    "height": 1,
                    "marginVertical": 5,
                    "width": "95%",
                  }
                }
              />
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderRadius": 25,
                      "flexDirection": "row",
                      "marginBottom": 20,
                      "overflow": "hidden",
                      "width": "90%",
                    },
                    {
                      "backgroundColor": "#E0E0E0",
                    },
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#668F80",
                      "flex": 1,
                      "justifyContent": "center",
                      "paddingVertical": 20,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#FFF",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Déconnecter
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="#668F80"
        color="#fff"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Options"
        titleColor="#FFF"
        titleFontSize={24}
        titleFontWeight="bold"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
